admin/utils.go

```diff
@@ -49,6 +49,14 @@ func IsNil(i interface{}) bool {
 	return false
 }
 
+// SetOrDefault returns the address of the given value or nil if it equals defaultValue
+func SetOrDefault[T comparable](val T, defaultValue T) *T {
```

Comment:
@wtrocki nit: I chose this name to by symmetrical with `GetOrDefault`. However, the more I think about it, `SetOrNil` actually makes more sense?

Reply:
Done